### ISSUE TITLE
[a11y] esc key should always close modals

### ISSUE THREAD

**Issue by vickiniu** [freedomofpress/securedrop-client#3503]
When a user keyboard navs away from a modal, there is not a good way to re-focus the modal state, and thus they are unable to close the modal. Should ensure that all modals are closable with the `esc` key.

Since we have a number of modals across the Inbox, may also be worth wrapping some of the accessibility features into an internal `Modal` utility that we can import + use

**Comment by zenmonkeykstop** [comment]
The Ant modal already defaults to Esc=close unless you disable it. And if they can nav away from it, how is it a modal? This sounds more like an issue with keyboard navigation overriding the UI defaults?
